runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java

```diff
@@ -27,28 +29,14 @@
 public class MultiObjectSMREntry extends LogEntry implements ISMRConsumable {
 
     // map from stream-ID to a list of updates encapsulated as MultiSMREntry
+    //TODO(Maithem): we should hide entryMap
     @Getter
-    public Map<UUID, MultiSMREntry> entryMap = Collections.synchronizedMap(new HashMap<>());
+    public Map<UUID, List<SMREntry>> entryMap = Collections.synchronizedMap(new HashMap<>());
```

Comment:
@WenbinZhu is this what you meant when you talked about collapsing MultiSMREntry and MultiObjectSMREntry ?

Reply:
Not really, the collapsing I was referring to is to use `MultiObjectSMREntry` for both transaction and non-transaction, instead of the current `MultiObjectSMREntry` for transaction and, `SMREntry` for non-transaction and `MultiSMREntry` for checkpoint. But yes, one result of this kind of collapsing is `MultiSMREntry` will be completely removed.